src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -253,8 +248,15 @@ class DeepDiveLogParser extends JavaTokenParsers {
       }
     )
 
-  def functionMode = "mode" ~> "=" ~> functionModeType
-  def inferenceMode = "mode" ~> "=" ~> inferenceModeType
+  def ruleAnnotation = "@" ~> annotationName ~ ("(" ~> rep1sep(ident, ",") <~ ")") ^^ {
+    case (name ~ args) => RuleAnnotation(name, args)
+  }
+
+  def ruleAnnotations = rep(ruleAnnotation)
```

Comment:
I was checking which issues this PR was closing, and it seems the ruleAnnotations aren't really being parsed or attached to the parse tree yet, right?